### PR TITLE
[#712] Add legionctl command generate-token

### PR DESCRIPTION
**Pull request by skayerok**

### CODE REVIEW

**Comment by skayerok** [comment]
This closes #712

File: legion/bin/legionctl [hunk 1]
```diff
@@ -60,6 +60,15 @@ if __name__ == '__main__':
     add_edi_arguments(login_parser, required=True)
     login_parser.set_defaults(func=login)
 
+    # --------- SECURITY SECTION -----------
```
Comment (mcdoker18) [inline legion/bin/legionctl]: Pleases add the login parser to the security section and remove login section.

File: legion/docs/source/legionctl.md [hunk 2]
```diff
@@ -32,4 +32,11 @@ Another option save the token in config file. Further you can use legionctl comm
...
 legionctl inspect
-```
\ No newline at end of file
+```
+
+##Model
+
+To get json web model token for a model, you can use a command:
+```bash
+legionctl generate-token --edi <edi-url> --token <your-token> --model-id <model-id> --model-version <model-version>
```
Comment (mcdoker18) [inline legion/docs/source/legionctl.md]: Remove `--token <your-token>`. This parameter is optional, and it can confuse here.

**Comment by skayerok** [comment]
bot, test all

File: tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot [hunk 3]
```diff
@@ -207,4 +207,22 @@ Login. Override login values
 
     ${res}=  Shell  LEGION_CONFIG=${LOCAL_CONFIG} legionctl --verbose inspect --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --token wrong-token
              Should not be equal  ${res.rc}  ${0}
-             Should contain       ${res.stderr}  Credentials are not correct
\ No newline at end of file
+             Should contain       ${res.stderr}  Credentials are not correct
+
+Get token from EDI
+    [Documentation]  Try to get token from EDI
+    [Setup]   Run EDI deploy and check model started  ${MODEL_TEST_ENCLAVE}  ${TEST_MODEL_IMAGE_5}  ${TEST_COMMAND_MODEL_ID}  ${TEST_MODEL_5_VERSION}
```
Comment (mcdoker18) [inline tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: I guess we don't need to deploy and undeploy a model for this test.

File: legion/legion/edi/security.py [hunk 4]
```diff
@@ -134,3 +134,14 @@ def get_security_params_from_config():
             LOG.debug('Exception during parsing of legion config {}'.format(e), exc_info=True)
 
     return {}
+
```
Comment (mcdoker18) [inline legion/legion/edi/security.py]: Also, please replace the http request by legionctl command:
* https://github.com/legion-platform/legion/blob/develop/tests/robot/resources/keywords.robot#L182

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

File: tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot [hunk 5]
```diff
@@ -207,4 +207,20 @@ Login. Override login values
...
+
+    ${res} =  Shell  legionctl generate-token --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --model-id ${TEST_COMMAND_MODEL_ID} --token "${DEX_TOKEN}"
+              Should not be equal  ${res.rc}  ${0}
+
+    ${res} =  Shell  legionctl generate-token --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --model-version ${TEST_MODEL_5_VERSION} --token "${DEX_TOKEN}"
+              Should not be equal  ${res.rc}  ${0}
+
+    ${res} =  Shell  legionctl generate-token --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --model-id ${TEST_COMMAND_MODEL_ID} --model-version ${TEST_MODEL_5_VERSION} --token "invalid-token"
+              Should not be equal  ${res.rc}  ${0}
+
```
Comment (kirillmakhonin) [inline tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: Maybe we have to add checking of error messages?

Comment (skayerok) [reply tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: Fixed
Also removed test case with invalid token, which is already tested in robot tests 1.0

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

File: tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot [hunk 6]
```diff
@@ -207,4 +207,18 @@ Login. Override login values
...
+              Should be equal       ${res.rc}  ${0}
+              Should not be empty   ${res.stdout}
+
+    ${res} =  Shell  legionctl generate-token --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --model-id ${TEST_COMMAND_MODEL_ID} --token "${DEX_TOKEN}"
+              Should not be equal   ${res.rc}  ${0}
+              Should contain        ${res.stderr}  Requested field model_version is not set
+
+    ${res} =  Shell  legionctl generate-token --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --model-version ${TEST_MODEL_5_VERSION} --token "${DEX_TOKEN}"
+              Should not be equal   ${res.rc}  ${0}
+              Should contain        ${res.stderr}  Requested field model_id is not set
```
Comment (kirillmakhonin) [inline tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: Where is a test with incorrect dex token?

Comment (skayerok) [reply tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: For example, there is a test `Login. Wrong token`.
It tests different command, but inside this command calls the same function `build_client`.

But if you think that this test should also be here, I can revert it back.

Comment (kirillmakhonin) [reply tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: @mcdoker18 what do you think?

**Comment by skayerok** [comment]
bot, test all

File: tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot (hunk 6 above)
Comment (skayerok) [reply tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: Revert back this test case

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

File: tests/robot/tests/3_edi_one_model/3.0_check_edi_one_model.robot [hunk 7]
```diff
@@ -21,48 +21,6 @@ Check EDI availability in all enclaves
     \                       Should Be Equal As Integers   ${edi_state.rc}     0
     [Teardown]              NONE
 
-Get token from EDI with valid parameters
```
Comment (dsuslov) [inline tests/robot/tests/3_edi_one_model/3.0_check_edi_one_model.robot]: @skayerok why did you delete tests for RESTful API? legionctl is not gonna be a single client for this API.

File: tests/robot/resources/keywords.robot [hunk 8]
```diff
@@ -177,14 +177,12 @@ Build enclave EDGE URL
 
 Get token from EDI
     [Documentation]  get token from EDI for the EDGE session
-    [Arguments]     ${enclave}   ${model_id}   ${model_version}
```
Comment (dsuslov) [inline tests/robot/resources/keywords.robot]: @skayerok why did you delete tests for RESTful API? legionctl is not gonna be a single client for this API. Keep test for RESTful API too

Comment (skayerok) [reply tests/robot/resources/keywords.robot]: Ok, reverted back these tests

**Comment by skayerok** [comment]
bot, test all
